Build error of current `master` caused by `haddock-library-1.11.0`

Current `master` (c84f467c2453e59f6bfd24f06ee3615befddcb18) fails to build with GHC 9.2.4 in the latest ecosystem:
```
$ cabal build
Build profile: -w ghc-9.2.4 -O1
...
[ 31 of 181] Compiling Distribution.Server.Pages.Package.HaddockHtml ( ... )

src/Distribution/Server/Pages/Package/HaddockHtml.hs:27:25: error:
    • No instance for (HTML (Int, Html))
        arising from a use of ‘ordList’
    • In the ‘markupOrderedList’ field of a record
      In the expression:
        Markup
          {markupEmpty = noHtml, markupString = toHtml,
           ...
           markupOrderedList = ordList, markupDefList = defList,
          ...
           markupTable = \ (Table h r) -> makeTable h r}
      In an equation for ‘htmlMarkup’: ...
         
   |
27 |   markupOrderedList   = ordList,
   |                         ^^^^^^^
```
This is the used dependencies according to `cabal-plan`:
```
  Cabal-3.6.3.0
  HStringTemplate-0.8.8
  HTTP-4000.3.16
  QuickCheck-2.14.2
  array-0.5.4.0
  base-4.15.1.0
  binary-0.8.8.0
  blaze-builder-0.4.2.2
  blaze-html-0.9.1.2
  base16-bytestring-1.0.2.0
  base64-bytestring-1.2.1.0
  bytestring-0.10.12.1
  acid-state-0.16.1.1
  commonmark-0.2.2
  commonmark-extensions-0.2.3.2
  containers-0.6.4.1
  cereal-0.5.8.3
  cryptohash-md5-0.11.101.0
  cryptohash-sha256-0.11.102.1
  csv-0.1.2
  ed25519-0.0.5.0
  deepseq-1.4.5.0
  directory-1.3.6.2
  filepath-1.4.2.1
  fail-4.9.0.0
  hackage-security-0.6.2.2
  hackage-security-HTTP-0.1.1.1
  haddock-library-1.11.0
  happstack-server-7.7.2
  hashable-1.4.1.0
  hslogger-1.3.1.0
  lifted-base-0.2.3.12
  mime-mail-0.5.1
  mtl-2.2.2
  unordered-containers-0.2.19.1
  network-3.1.2.7
  network-bsd-2.8.1.0
  network-uri-2.6.4.1
  parsec-3.1.14.0
  pretty-1.1.3.6
  process-1.6.13.2
  random-1.2.1.1
  rss-3000.2.0.7
  scientific-0.3.7.0
  safecopy-0.10.4.2
  semigroups-0.19.2
  aeson-2.0.3.0
  snowball-1.0.0.1
  split-0.2.3.5
  stm-2.5.0.0
  async-2.2.4
  text-1.2.5.0
  tagged-0.8.6.1
  time-1.9.3
  tokenize-0.3.0
  tar-0.5.1.1
  transformers-0.5.6.2
  attoparsec-0.14.4
  attoparsec-iso8601-1.0.2.1
  unix-2.7.2.2
  vector-0.12.3.1
  xhtml-3000.2.2.1
  xmlgen-0.6.2.2
  xss-sanitize-0.3.7
  zlib-0.6.3.0
```

--- a/.github/workflows/haskell-ci.yml
+++ b/.github/workflows/haskell-ci.yml
@@ -8,9 +8,9 @@
 #
 # For more information, see https://github.com/haskell-CI/haskell-ci
 #
-# version: 0.15.20220710
+# version: 0.15.20220822
 #
-# REGENDATA ("0.15.20220710",["github","hackage-server.cabal"])
+# REGENDATA ("0.15.20220822",["github","hackage-server.cabal"])
 #
 name: Haskell-CI
 on:
@@ -34,9 +34,9 @@ jobs:
     strategy:
       matrix:
         include:
-          - compiler: ghc-9.2.3
+          - compiler: ghc-9.2.4
             compilerKind: ghc
-            compilerVersion: 9.2.3
+            compilerVersion: 9.2.4
             setup-method: ghcup
             allow-failure: false
           - compiler: ghc-9.0.2
@@ -62,7 +62,7 @@ jobs:
           apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5
           if [ "${{ matrix.setup-method }}" = ghcup ]; then
             mkdir -p "$HOME/.ghcup/bin"
-            curl -sL https://downloads.haskell.org/ghcup/0.1.17.8/x86_64-linux-ghcup-0.1.17.8 > "$HOME/.ghcup/bin/ghcup"
+            curl -sL https://downloads.haskell.org/ghcup/0.1.18.0/x86_64-linux-ghcup-0.1.18.0 > "$HOME/.ghcup/bin/ghcup"
             chmod a+x "$HOME/.ghcup/bin/ghcup"
             "$HOME/.ghcup/bin/ghcup" install ghc "$HCVER" || (cat "$HOME"/.ghcup/logs/*.* && false)
             "$HOME/.ghcup/bin/ghcup" install cabal 3.6.2.0 || (cat "$HOME"/.ghcup/logs/*.* && false)
@@ -73,7 +73,7 @@ jobs:
             apt-get update
             apt-get install -y "$HCNAME" libbrotli-dev
             mkdir -p "$HOME/.ghcup/bin"
-            curl -sL https://downloads.haskell.org/ghcup/0.1.17.8/x86_64-linux-ghcup-0.1.17.8 > "$HOME/.ghcup/bin/ghcup"
+            curl -sL https://downloads.haskell.org/ghcup/0.1.18.0/x86_64-linux-ghcup-0.1.18.0 > "$HOME/.ghcup/bin/ghcup"
             chmod a+x "$HOME/.ghcup/bin/ghcup"
             "$HOME/.ghcup/bin/ghcup" install cabal 3.6.2.0 || (cat "$HOME"/.ghcup/logs/*.* && false)
           fi

--- a/cabal.haskell-ci
+++ b/cabal.haskell-ci
@@ -14,3 +14,8 @@ installed: +all -Cabal -text -parsec
 distribution: focal
 
 apt: libbrotli-dev
+
+-- Make sure the haddock step is included,
+-- even though we don't define any library.
+haddock-components: all
+  -- since haskell-ci 0.15.20220822

--- a/hackage-server.cabal
+++ b/hackage-server.cabal
@@ -118,6 +118,9 @@ common defaults
   build-depends:
     , aeson                 ^>= 2.0.3.0
     , Cabal                 ^>= 3.6.3.0
+    , Cabal-syntax          ^>= 3.6.0.0
+        -- Cabal-syntax needs to be bound to constrain hackage-security
+        -- see https://github.com/haskell/hackage-server/issues/1130
     , fail                  ^>= 4.9.0
       -- we use Control.Monad.Except, introduced in mtl-2.2.1
     , network               >=  3 && < 3.2
@@ -390,9 +393,14 @@ library lib-server
     , cryptohash-sha256     ^>= 0.11.100
     , csv                   ^>= 0.1
     , ed25519               ^>= 0.0.5
-    , hackage-security      ^>= 0.6
+    , hackage-security       >= 0.6 && < 0.7
+        -- N.B: hackage-security-0.6.2 uses Cabal-syntax-3.8.1.0
+        -- see https://github.com/haskell/hackage-server/issues/1130
+        -- Thus, we need to include Cabal-syntax as dependency explicitly
     , hackage-security-HTTP ^>= 0.1.1
-    , haddock-library       > 1.7 && < 2
+    , haddock-library        >= 1.7.0 && < 1.11
+        -- haddock-library-1.11.0 changed type of markupOrderedList
+        -- see https://github.com/haskell/hackage-server/issues/1128
     , happstack-server      ^>= 7.7.1
     , hashable              ^>= 1.3 || ^>= 1.4
     , hslogger              ^>= 1.3.1

--- a/hackage-server.cabal
+++ b/hackage-server.cabal
@@ -27,7 +27,7 @@ copyright:    2008-2015 Duncan Coutts,
 license:      BSD-3-Clause
 license-file: LICENSE
 
-tested-with: GHC == { 9.2.3, 9.0.2, 8.10.7, 8.8.4 }
+tested-with: GHC == { 9.2.4, 9.0.2, 8.10.7, 8.8.4 }
 
 data-dir: datafiles
 data-files:
@@ -111,12 +111,12 @@ common defaults
     , process                >= 1.6   && < 1.7
     , text                  ^>= 1.2.5.0 || ^>= 2.0
     , time                   >= 1.9   && < 1.13
-    , transformers           >= 0.5   && < 0.6
-    , unix                   >= 2.7   && < 2.8
+    , transformers           >= 0.5   && < 0.7
+    , unix                   >= 2.7   && < 2.9
     , scientific
   -- other dependencies shared by most components
   build-depends:
-    , aeson                 ^>= 2.0.3.0
+    , aeson                 ^>= 2.0.3.0 || ^>= 2.1.0.0
     , Cabal                 ^>= 3.6.3.0
     , Cabal-syntax          ^>= 3.6.0.0
         -- Cabal-syntax needs to be bound to constrain hackage-security
@@ -129,7 +129,7 @@ common defaults
     , parsec                ^>= 3.1.13
     , tar                   ^>= 0.5
     , unordered-containers  ^>= 0.2.10
-    , vector                ^>= 0.12
+    , vector                ^>= 0.12 || ^>= 0.13.0.0
     , zlib                  ^>= 0.6.2
 
   ghc-options: -Wall -fwarn-tabs -fno-warn-unused-do-bind -fno-warn-deprecated-flags -funbox-strict-fields
@@ -377,7 +377,7 @@ library lib-server
     , async                 ^>= 2.2.1
     -- requires bumping http-io-streams
     , attoparsec            ^>= 0.14.4
-    , attoparsec-iso8601    ^>= 1.0
+    , attoparsec-iso8601    ^>= 1.0       || ^>= 1.1.0.0
     , base16-bytestring     ^>= 1.0
     -- requires bumping http-io-streams
     , base64-bytestring     ^>= 1.2.1.0
@@ -401,15 +401,15 @@ library lib-server
     , haddock-library        >= 1.7.0 && < 1.11
         -- haddock-library-1.11.0 changed type of markupOrderedList
         -- see https://github.com/haskell/hackage-server/issues/1128
-    , happstack-server      ^>= 7.7.1
-    , hashable              ^>= 1.3 || ^>= 1.4
+    , happstack-server      ^>= 7.7.1     || ^>= 7.8.0
+    , hashable              ^>= 1.3       || ^>= 1.4
     , hslogger              ^>= 1.3.1
     , lifted-base           ^>= 0.2.1
     , mime-mail             ^>= 0.5
     , random                ^>= 1.2
     , rss                   ^>= 3000.2.0.7
     , safecopy              ^>= 0.10
-    , semigroups            ^>= 0.19
+    , semigroups            ^>= 0.20
     , split                 ^>= 0.2
     , stm                   ^>= 2.5.0
     , tagged                ^>= 0.8.5

